src/Facebook/FileUpload/FacebookVideoUploader.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\FileUpload;
+
+use Facebook\Exceptions\FacebookResponseException;
+use Facebook\Exceptions\FacebookSDKException;
+use Facebook\Facebook;
+
+/**
+ * Class VideoUploadContext
+ * @package Facebook
+ */
+class VideoUploadContext
+{
+    /**
+     * @var string
+     */
+    public $accessToken;
+
+    /**
+     * @var string Video upload target Id, could be page_id || user_id || event_id || group_id
+     */
+    public $targetId;
+
+    /**
+     * @var string Upload file path
+     */
+    public $filePath;
+
+    /**
+     * @var int Upload file size in bytes.
+     */
+    public $fileSize;
+
+    /**
+     * @var int ID of the upload session
+     */
+    public $uploadSessionId;
+
+    /**
+     * @var int Start byte position of the next file chunk
+     */
+    public $startOffset;
+
+    /**
+     * @var int End byte position of the next file chunk
+     */
+    public $endOffset;
+
+    /**
+     * @var int ID of the video
+     */
+    public $videoId;
+}
+
+/**
+ * Class FacebookVideoUploader.
+ */
+class FacebookVideoUploader
+{
+    /**
+     * @var VideoUploadContext
+     */
+    protected $context;
+
+    /**
+     * @var int Max retry times
+     */
+    protected $maxTransferTries;
+
+    /**
+     * @var Facebook Reference to Facebook
+     */
+    protected $facebook;
```

Comment:
Instead of referring Facebook instance directly, could we refer a http client do send the request as 
 https://github.com/facebook/facebook-php-sdk-v4/blob/4.0-dev/src/Facebook/FacebookRequest.php#L92
@SammyK , will that be ok for you?